docs/src/pages/components/data-grid/layout/AutoHeightGrid.tsx

```diff
@@ -1,18 +1,28 @@
 import * as React from 'react';
+import Button from '@mui/material/Button';
 import { DataGrid } from '@mui/x-data-grid';
 import { useDemoData } from '@mui/x-data-grid-generator';
 
 export default function AutoHeightGrid() {
+  const [nbRows, setNbRows] = React.useState(5);
+  const removeRow = () => setNbRows((x) => Math.max(0, x - 1));
+  const addRow = () => setNbRows((x) => Math.min(100, x + 1));
+
   const { data } = useDemoData({
     dataSet: 'Commodity',
-    rowLength: 4,
+    rowLength: 100,
     maxColumns: 6,
   });
 
   return (
-    <div style={{ height: 400, width: '100%' }}>
-      <DataGrid autoHeight {...data} />
-      <p>more content</p>
```

Comment:
For context, I added this node because the first implement of autoHeight got it wrong, `<p>more content</p>` we displayed on top of the data grid 😆. I used the visual regression test for this, I'm not sure there are any other tests: https://github.com/mui-org/material-ui-x/pull/981#discussion_r570859839. I guess we can wait for a regression that might never happen to reconsider the approach.